### DISCUSSION TITLE
I accidentally cleared the canvas, is there a way to restore the previous drawing?

### DISCUSSION THREAD

**Discussion by borekb** [excalidraw/excalidraw#6648] [Q&A]
As the title says.

Another day it happened to me that I clicked the permalink and didn't realize I have unsaved drawing in Excalidraw already.

In such cases, is there a way to restore the unsaved drawing from somewhere?

**Comment by dwelle** [comment]
Hey. Sadly, no. We're working on making it more clear and allowing you to save/export before replacing the content, and further down the line we'll make shareable links not override the canvas at all, similar to collab rooms. But in your cases, nothing we can do now :(.

**Reply by borekb** [reply]
I wouldn't mind digging into local storage or something like that, would it be possible to cache/store previous drawings somewhere instead of entirely replacing them?

**Reply by pravindev21** [reply]
I also had the same bad experience, accidently i drag the file from browser download popup to the canvas, it didn't got any warning for the replacement. @dwelle please address this as bug / feature, so in future developer not affected by this accidental issues.

**Reply by StarkPrince** [reply]
I have cleared the browser cache and local storage and got to notice that my hours of works for the ER diagram have gone. Luckily, as per client request, I had exported that as PNG, but even after converting that to SVG, I am not able to achieve the previous state of files.

**Reply by prahulk46a** [reply]
Just lost my 1+years drawings😭

**Reply by andreanerla** [reply]
same

**Reply by sraka** [reply]
this is such a bad experience for users, and should be addressed asap by the developers.
atleast give a warning that there are unsaved changes, or a confirmation dialog to ask user when they drag and drop a local file.

**Reply by myselfshravan** [reply]
There is a way to recover it from LDB chrome local store history. I did some research on claude and I was able to go deeper in the macOS file versioning of the browser cache and recovered completely.

**Comment by DETROJAJAY** [comment]
I faced same issue. still is there no any solution like get again from catch or cookie ?
